service/src/test/org/apache/hive/service/cli/operation/TestQueryLifeTimeHooksWithSQLOperation.java

```diff
@@ -110,6 +118,45 @@ public void afterExecution(QueryLifeTimeHookContext ctx, boolean hasError) {
       assertNull(ctx.getHookContext().getException());
       assertNotNull(ctx.getHookContext().getQueryInfo());
       assertNotNull(ctx.getHookContext().getQueryInfo().getQueryDisplay());
+      assertQueryId(ctx.getQueryId());
     }
   }
+
+  /**
+   * Asserts that the specified query id exists and has the expected prefix and size.
+   *
+   * <p>
+   * A query id looks like below:
+   * <pre>
+   *   username_20220330093338_dab90f30-5e79-463d-8359-0d2fff57effa
+   * </pre>
+   * and we can accurately predict how the prefix should look like. T
+   * </p>
+   *
+   * @param actualQueryId the query id to verify
+   */
+  private static void assertQueryId(String actualQueryId) {
+    assertNotNull(actualQueryId);
+    String expectedIdPrefix = makeQueryIdStablePrefix();
+    String actualIdPrefix = actualQueryId.substring(0, expectedIdPrefix.length());
+    assertEquals(expectedIdPrefix, actualIdPrefix);
+    assertEquals(expectedIdPrefix.length() + 41, actualQueryId.length());
+  }
+
+  /**
+   * Makes a query id prefix that is stable for an hour. The prefix changes every hour but this is enough to guarantee
```

Comment:
What happens if the test starts at let's say 2022.03.31 01:59:59.999 and this method is called at 2022.03.31 02:00:00.000 ?

Does the query id format should be checked here? How about adding an UT for the query generator itself.

Reply:
To avoid the risk we can remove the prefix check and keep only the length check. Alternatively, we can simply `assertNotNull` for the queryId and nothing more. What do you prefer?

Reply:
IUUC  `TestQueryLifeTimeHooksWithSQLOperation` tests whether data is passed to  `QueryLifeTimeHookWithParseHooks`. I think the test is more isolated and focus only to the data broadcasting functionality if the data is generated by the test. It can be achieved by setting the queryId explicitly in the beginning of the test to some constant and check it in the assertion part the exact same constant. I believe it can be set by `hive.query.id` config setting.

Or you can also ask the actual queryId via the `hive.query.id` config setting somewhere in the beginning of the test and use that value in the assert part instead of generating a new one.

Or as a last option if none of above works a regex can be used to check the format.

If you also want to test the queryId format is correct an UT for `QueryPlan.makeQueryId` would do that but I think a regex is still needed for that or at least for checking the date time part.

Reply:
@kasakrisz Thanks for pointing out the `hive.query.id` property, I wasn't aware of it. It seems that if we have this property then we can obtain the queryId in before/after hook methods by querying the configuration. Apparently we don't necessarily need the new API introduced in this PR (apart from convenience) so I am inclined to close this without merging. WDYT?

Reply:
If getting the queryId from conf is sufficient the new API is not necessarily needed. I think sharing data via config between modules is not a nice solution but that would be a different project to get rid of that.